packages/muelu/src/Graph/MatrixTransformation/MueLu_AmalgamationInfo_def.hpp

```diff
@@ -58,21 +58,58 @@
 
 #include "MueLu_AmalgamationInfo_decl.hpp"
 #include "MueLu_Exceptions.hpp"
+
 #include "MueLu_Aggregates.hpp"
+#include "MueLu_Aggregates_kokkos.hpp"
 
 namespace MueLu {
 
   template <class LocalOrdinal, class GlobalOrdinal, class Node>
-  void AmalgamationInfo<LocalOrdinal, GlobalOrdinal, Node>::UnamalgamateAggregates(const Aggregates& aggregates,
-        Teuchos::ArrayRCP<LocalOrdinal>& aggStart, Teuchos::ArrayRCP<GlobalOrdinal>& aggToRowMap) const {
-    const int myPid = aggregates.GetMap()->getComm()->getRank();
-    Teuchos::ArrayView<const GO> nodeGlobalElts = aggregates.GetMap()->getLocalElementList();
-    Teuchos::ArrayRCP<LO> procWinner   = aggregates.GetProcWinner()->getDataNonConst(0);
-    Teuchos::ArrayRCP<LO> vertex2AggId = aggregates.GetVertex2AggId()->getDataNonConst(0);
+  void AmalgamationInfo<LocalOrdinal, GlobalOrdinal, Node>::
+  UnamalgamateAggregates(const Aggregates& aggregates,
+                         Teuchos::ArrayRCP<LocalOrdinal>& aggStart,
+                         Teuchos::ArrayRCP<GlobalOrdinal>& aggToRowMap) const {
+
+    UnamalgamateAggregates(aggregates.GetMap(),
+                           aggregates.GetProcWinner(),
+                           aggregates.GetVertex2AggId(),
+                           aggregates.GetNumAggregates(),
+                           aggStart,
+                           aggToRowMap);
+
+  } //UnamalgamateAggregates
+
+  template <class LocalOrdinal, class GlobalOrdinal, class Node>
+  void AmalgamationInfo<LocalOrdinal, GlobalOrdinal, Node>::
+  UnamalgamateAggregates(const Aggregates_kokkos& aggregates,
+                         Teuchos::ArrayRCP<LocalOrdinal>& aggStart,
+                         Teuchos::ArrayRCP<GlobalOrdinal>& aggToRowMap) const {
+
+    UnamalgamateAggregates(aggregates.GetMap(),
+                           aggregates.GetProcWinner(),
+                           aggregates.GetVertex2AggId(),
+                           aggregates.GetNumAggregates(),
+                           aggStart,
+                           aggToRowMap);
+
+  } //UnamalgamateAggregates
```

Comment:
This seems a little gross to see this duplication in a few places, but I guess it's necessary until we merge `Aggregates` and `Aggregates_kokkos`?

Reply:
I could have made Aggregates a parent class for Aggregates_kokkos, but I figured you'd get rid of it anyway soon.